src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Applications/ApplicationsEndpoints.cs

```diff
@@ -35,10 +35,15 @@ public void Register(IEndpointRouteBuilder endpointRouteBuilder)
         .WithOpenApi("Submit an application", string.Empty, "application_post")
         .RequireAuthorization();
 
-    endpointRouteBuilder.MapGet("/api/applications", async (ApplicationStatus[]? byStatus, IMessageBus messageBus, IMapper mapper) =>
+    endpointRouteBuilder.MapGet("/api/applications/{id?}", async (string? id, ApplicationStatus[]? byStatus, HttpContext ctx, IMessageBus messageBus, IMapper mapper) =>
         {
+          var userId = ctx.User.GetUserContext()?.UserId;
+
+          bool isNotGuid = !Guid.TryParse(id, out _); if (isNotGuid) { id = null; }
```

Comment:
In our current situation, we encounter an issue where the client occasionally transmits an undefined or empty string as an ID. Since these values are not null, attempting to parse them into a Guid results in an exception. To mitigate this, I have implemented a temporary solution. For a more robust and permanent fix, I plan to introduce an AutoMapper resolver in the upcoming refactor. This resolver will proactively validate the property, ensuring it's in a suitable format before we attempt to parse it as a Guid during the mapping process.